contrib/mesos/pkg/scheduler/plugin_test.go

```diff
@@ -200,44 +201,56 @@ func (lw *MockPodsListWatch) Pod(name string) *api.Pod {
 	return nil
 }
 func (lw *MockPodsListWatch) Add(pod *api.Pod, notify bool) {
-	lw.lock.Lock()
-	defer lw.lock.Unlock()
+	func() {
+		lw.lock.Lock()
+		defer lw.lock.Unlock()
+		lw.list.Items = append(lw.list.Items, *pod)
+	}()
```

Comment:
I think it's a good common pattern: don't lock anything when you call code which might lock other stuff. Never lock longer than needed. It's not bad that the channel behind the fakeWatch is invisible, it's bad that we keep a lock while calling unknown code.